packages/mds-cache/index.ts

```diff
@@ -198,10 +198,28 @@ async function hwrite(suffix: string, item: CacheReadDeviceResult | Telemetry |
     await (await getClient()).hdelAsync(key, ...nulls)
   }
 
-  await (await getClient()).hmsetAsync(key, hmap)
+  const client = await getClient()
+
+  if (suffix === 'event') {
+    await client.hmsetAsync(`provider:${item.provider_id}:latest_event`, hmap)
+  }
+  await client.hmsetAsync(key, hmap)
```

Comment:
Might be able to do these in parallel using `Promise.all([...])` if that's of interest.

Reply:
Only thing that I see could be kind of challenging is that one of the writes is conditional. Do you know if you can embed logic into a `Promise.all()`? Something like... `Promise.all([client.hmsetAsync(key, hmap), suffix === client ? client.hmsetAsync(`provider...etc...`) : null])`

Reply:
```ts
await Promise.all((suffix === 'event' ? [`provider:${item.provider_id}:latest_event`, key] : [key]).map(k => client.hmsetAsync(k, hmap)))
```